src/sentry/snuba/tasks.py

```diff
@@ -218,16 +218,26 @@ def _create_in_snuba(subscription: QuerySubscription) -> str:
             snuba_query,
             subscription.project.organization_id,
         )
-        # TODO: determine whether concatenating query_extra is proper
+        params = {
+            "organization_id": subscription.project.organization_id,
+            "project_id": [subscription.project_id],
+        }
+        skip_time_conditions = True
+        if subscription.timebox_start:
+            start = subscription.timebox_start.isoformat()
+            window = snuba_query.time_window
+            end = (subscription.timebox_start + window).isoformat()
+            skip_time_conditions = False
+            params[start] = start
+            params[end] = end
```

Comment:
Should we just be setting the data as start / end times instead of start / duration? that way we only calculate on write vs read.

Reply:
So - this is 'read' of the QuerySubscription - but it's 'write' of the snql query

The 'end' will always be a derivative of the SnubaQuery.time_window - so there's not really any reason to duplicate that value.
Part of my reason to derive the end here rather than set it on the QuerySubscription is _also_ to ensure that we never introduce a query with too large of a window.
This is the reason why the time conditions was being skipped originally as well

Reply:
hmmm....
I guess i'll actually just ave it to the model as well :/